### ISSUE TITLE
weather binding and my.openhab doesn't work

### ISSUE THREAD

**Issue by staehler** [openhab/openhab#2034]
when connected to openhab via my.openhab (habdroid) the weather binding webview (url="/weather?locationId=in&layout=weather&iconset=colorful") just shows "unauthorized".

**Comment by teichsta** [comment]
could you please have a look at this @gerrieg? Thanks, Thomas E.-E.

**Comment by adrianmihai83** [comment]
Same thing happens to me, is there a solution to this?

screenshot attached
![screenshot_2015-09-14-17-48-26](https://cloud.githubusercontent.com/assets/8840601/9852863/2a371c28-5b09-11e5-833a-ee8329e3fa58.png)
![screenshot_2015-09-14-17-47-32](https://cloud.githubusercontent.com/assets/8840601/9852864/2a40acb6-5b09-11e5-871a-c143c76f19a4.png)

**Comment by staehler** [comment]
No, it seems there is no solution available ... unfortunately.

**Comment by adrianmihai83** [comment]
This is a problem I encountered in previous builds also, to be more precise it didn't work for me in any of the builds. There are other things concerning me, other problems that make this nice opensource to be useless and that is the case of the missing icons in HabDroid when using myopenhab. And the most frustrating thing is that nobody answers, either way, is this thing checked out, is it not...is it left to be forgotten in the never ending new issues... Or maybe it is not worthy of a feature...

I have to roll back the runtime to get it working decently, there is an urge to add to the build numbers, add complexity with the expence of working things. I had used other software for the last 9 mounths and decided to come back to check things out. The thing that got me to come back is, and I quote here: "Well, the openHAB 1.x core runtime has been very stable and thus did not see much evolution during the last two releases. We will keep it this way and will only concentrate on security fixes and critical patches from now on." Instantly I thought, well, this is a mature software, everything  has been fixed, time to use it propertly... Now, the greatest feature, myopenhab - to be more precise, the integration of the dedicated mobile application with myopenhab service - is useless...

But hey, you can't be upset, it's opensource, as everybody will say: use it or leave it!

**Comment by eschizoid** [comment]
@gerrieg is there any update regarding to this?

Seems to be related to this:

https://github.com/openhab/openhab.android/issues/127#issuecomment-202026815

**Comment by gerrieg** [comment]
I don't know how to fix this in the weather binding. The weather URL is relative, so the weather URL in my.openhab a device is calling is my.openhab.org/weather. That URL might not be forwarded to the openhab server.

**Comment by watou** [comment]
Have you tried home.openHAB.org instead of my.openHAB. org?

**Comment by eschizoid** [comment]
@watou you meant to use that host as part of the webview url or change the url when using the external OAuth authentication?

**Comment by watou** [comment]
I recall @belovictor adding home.openhab.org to allow reverse proxying of all URLs on your openHAB server, so it wouldn't be a URL you used in your sitemap, just the URL to reach your server via home.openhab.org instead of my.openhab.org.

**Comment by eschizoid** [comment]
@watou Yeah, I just tried both approaches and still getting the `Unauthorized`

**Comment by watou** [comment]
Sorry, @eschizoid, out of ideas at the moment...

**Comment by eschizoid** [comment]
@watou, thanks for your help. I think we will have to wait got @belovictor and see wha he has to say regarding to this.

**Comment by jaydee73** [comment]
@belovictor : Any updates on this issue? Have you seen this already? (as you didn't reply to the last comment).

**Comment by digitaldan** [comment]
The problem is my.openhab only forwards requests to /rest/* and a few more "blessed" URL patterns, bindings that attach to other URL locations will not be forwarded.  One solution is to add "/weather*" to the allowed patterns.  Other solutions will probably require a larger refactoring.   With OH2 I have noticed a few missing locations, so I would vote to add this one as well .

**Comment by digitaldan** [comment]
@watou  do you want to assign this to me?  I will try to get this into a future release, no eta just yet on when that will be.

**Comment by watou** [comment]
Before it gets assigned, could you try the URL https://home.openhab.org/ instead of https://my.openhab.org?  I think I recall @belovictor added that so the service would forward all requests.

**Comment by eschizoid** [comment]
>Before it gets assigned, could you try the URL https://home.openhab.org/ instead of https://my.openhab.org? I think I recall @belovictor added that so the service would forward all requests.

@watou I did while ago a didn't made any difference

**Comment by jaydee73** [comment]
when I access the URL with home.openhab.org I'll get an "unauthorized" message in the webview frame (doesn't matter if I choose "ignore SSL certificate" or not). When I choose my.openhab.org, I'll get "Cannot get /weather?.....". So it make a difference, but only in case of different error messages. :)

**Comment by watou** [comment]
> with home.openhab.org I'll get an "unauthorized" message ...
> my.openhab.org, I'll get "Cannot get /weather?....."

@jaydee73 This is good information that might be useful to @belovictor regarding the "home.openhab.org" URL, in case it would be possible to allow the weather binding's URIs to be forwarded.

**Comment by botura** [comment]
Exactly the same errors here.
